perun-core/src/main/java/cz/metacentrum/perun/core/entry/ResourcesManagerEntry.java

```diff
@@ -361,6 +361,22 @@ public void removeGroupFromResources(PerunSession perunSession, Group group, Lis
 		return getResourcesManagerBl().getAssignedGroups(sess, resource);
 	}
 
+	public List<Group> getAssignedGroups(PerunSession sess, Resource resource, Member member) throws InternalErrorException, PrivilegeException, ResourceNotExistsException {
+		Utils.checkPerunSession(sess);
+		getResourcesManagerBl().checkResourceExists(sess, resource);
+
+		Facility facility = getPerunBl().getResourcesManagerBl().getFacility(sess, resource);
+
+		// Authorization
+		if (!AuthzResolver.isAuthorized(sess, Role.VOADMIN, resource) &&
+				!AuthzResolver.isAuthorized(sess, Role.VOOBSERVER, resource) &&
+				!AuthzResolver.isAuthorized(sess, Role.FACILITYADMIN, facility)) {
```

Comment:
I think you can use resource object there (it has facility ID as part of itself).